client.go

```diff
@@ -247,6 +248,19 @@ func (c *Client) getErrorFromResponse(resp *http.Response) (*errorObject, error)
 	return &s, nil
 }
 
+// Helper function to determine wither additional parameters should use ? or & to append args
+func getBasePrefix(basePath string) string {
+	var conjunction string
+	// check to see if there are already arguments in basePath (look for ? in path)
+	re := regexp.MustCompile(".*/[^/]*\\?[^/]*$")
+	if re.Match([]byte(basePath)) {
+		conjunction = "&"
+	} else {
+		conjunction = "?"
+	}
+	return fmt.Sprintf("%s%s", basePath, conjunction)
+}
```

Comment:
I think regular expressions make sense in systems where the filters / matches need to be user-supplied and cannot be known at build time. In this case, separate from my reuse comment, I'm not sure a regular expression is the right fit here. 

How would you feel about something like this? I've ran it against your proposed tests and got a :+1: :

```go
func getBasePrefix(basePath string) string {
	var p bool

	for _, c := range path.Base(basePath) {
		if c == '?' {
			p = true
			break
		}
	}

	if p {
		return basePath + "&"
	}

	return basePath + "?"
}
```

Reply:
I take your point on the regular expression, but is there a reason you didnt just use:

if strings.Contains(path.Base(basePath), "?")) { ... }

?

Reply:
![ryan reynolds confused](https://media0.giphy.com/media/gKsJUddjnpPG0/giphy.gif?cid=e3b0c442c4f045a3dd02c87f9abb3a042449a502c4554ca4&rid=giphy.gif)

Nope, wasn't thinking. 😄